### ISSUE TITLE
BitDefender is detecting inchi-1.exe as a virus

### ISSUE THREAD

**Issue by MikeWilliams-UK** [IUPAC-InChI/InChI#54]
I see this in my logs.

On-Access scanning has detected a threat. The file has been deleted.C:\Dev\Mike\***\Inchi\Inchi\inchi-1.exe is malware of type Gen:Variant.Lazy.591568

Please fix this

**Comment by giallu** [comment]
I think it's safe to say this is a false positive, maybe [open a report to Bitdefender](https://www.bitdefender.com/consumer/support/answer/29358/)?

**Comment by MikeWilliams-UK** [comment]
I agree that it's most likely a false posative, but it is also flagged by other engines too.

**Comment by MikeWilliams-UK** [comment]
Besides which it is not my place to report this, it should be done by the Inchi Trust

**Comment by fbaensch-beilstein** [comment]
I got a smiliary warning using G Data on Windows (sorry for the german):
Virus: Gen:Variant.Lazy.591568 (Engine A)
Datei: inchi-1.exe
Verzeichnis: [...]\InChI\INCHI-1-BIN\windows\64bit

**Comment by MikeWilliams-UK** [comment]
The binarys I used were downloaded from the release folder https://github.com/IUPAC-InChI/InChI/releases/download/v1.07.1/INCHI-1-BIN.zip

**Comment by MikeWilliams-UK** [comment]
And this for the 1.07 binaries @JanCBrammer 

https://github.com/IUPAC-InChI/InChI/releases/download/v1.07.0/INCHI-1-BIN.zip

**Comment by JanCBrammer** [comment]
> can you confirm the released binary files are built in a github workflow and not in any personal PC that may be compromised?

@giallu, the binaries under https://github.com/IUPAC-InChI/InChI/releases/download/ aren't built on GitHub runners. See https://github.com/IUPAC-InChI/InChI/issues/1.

As far a I know, currently, @djb-rwth, is building them on his machine.

**Comment by giallu** [comment]
Ok. I still think this is likely a false positive (otherwise more engines would mark the binaries as infected) but it make sense to start building them in the github enviroment so we can be pretty sure about it.

For my part, I can add to the cmake branch a package target that bundles the artifacts together and makes them available

**Comment by MikeWilliams-UK** [comment]
I think it would be a good idea to (code) sign all the binary files exe and dll in the release as that should help apease the AV vendors.

**Comment by fbaensch-beilstein** [comment]
I've uploaded the .exe to G Data, they accepted and whitelisted it.

**Comment by djb-rwth** [comment]
Hi all,
Thanks @fbaensch-beilstein for confirming my suspicions that this is a false positive.

This does not seem to be an isolated case in which `BitDefender` detected `Gen:Variant.Lazy.591568` falsely -- please refer to the following hyperlinks: [hl1](https://github.com/Norbyte/bg3se/issues/389), [hl2](https://www.reddit.com/r/antivirus/comments/1e6ik7y/is_genvariantlazy532905_a_false_positive/) or [hl3](https://community.bitdefender.com/en/discussion/90137/bitdefender-found-gen-variant-lazy-97932-in-a-dll).

Almost all AV software tends to be over-protective in cases of `.exe` files as malware and spyware most frequently sneak them within the OS.

Just recently, even the basic `Windows Security` falsely detected `PUABundler:Win32/Rostpay` within a `MyConsoleApplication.exe` which contained a compiled `Hello world` program and `PUA:Win32/Presenoker` inside `GCC 14.1` for `MS Windows` installation files. 

In line with @MikeWilliams-UK suggestion, all binaries will be digitally signed from now on and we shall see if that works.

I would like to encourage the users who encounter this sort of problem to submit the file(s) to the AV software HQs for further analyses/whitelisting, just like @fbaensch-beilstein did.

**Comment by djb-rwth** [comment]
Hi @MikeWilliams-UK,
All `Microsoft Windows` binaries have now been digitally signed in InChI `v1.07.2`, which has now been uploaded to [rwth](https://github.com/IUPAC-InChI/InChI/tree/rwth) branch.
Please note that `Microsoft Windows SDK's SignTool` has been used for digital signatures and that `Microsoft Powershell's` self-signed certificate has been used for creating digital certificates. 

Please be so kind as to test if your AV software still gives false positives.

**Comment by MikeWilliams-UK** [comment]
A virus total scan of inchi-1.exe in the rwh branch shows no false positive.
https://www.virustotal.com/gui/file/902ff29b4eb1e832982c966a3ef21be62f111436b95923e457c2bbc94d9253e7?nocache=1

However I do get a SmartScreen warning when I try to run it, which may concern some.
This may be due the cert being self signed or lack of reputation it's hard to tell.

We at [Chem4Word](https://www.Chem4Word.co.uk) use a code signing service run by the [.NET Foundation](https://dotnetfoundation.org/) which uses an EV certificate generated for us and stored in their Azure Key Vault.
See "Code signing" on https://dotnetfoundation.org/projects/benefits 

![image](https://github.com/user-attachments/assets/9e33c876-6245-4d76-a94b-55a29ca783dd)

**Comment by djb-rwth** [comment]
Hi @MikeWilliams-UK,
Thanks for the report.
I would say that the SmartScreen in question pops up because of the self-signed certificate which is mostly used for testing purposes.
I guess that we will have to consider using `DigiCert` or the `.NET Foundation` EV certificate, although we do not have the access to `Microsoft Azure` services at this point -- at least none that I am aware of.
Will keep you posted.

**Comment by MikeWilliams-UK** [comment]
If you wished to use the .NET foundations code signing service you would need to sign up/join their oganisation.

Details of benefit are here https://dotnetfoundation.org/membership/participation-types
I think the section "Open Source Maintainers / Project Owners" would be most appropriate

**Comment by MikeWilliams-UK** [comment]
There are examples of using this in Azure DevOps and GitHub actions on the main page of their CLI tool
https://github.com/dotnet/sign

**Comment by djb-rwth** [comment]
> There are examples of using this in Azure DevOps and GitHub actions on the main page of their CLI tool https://github.com/dotnet/sign

Hi @MikeWilliams-UK,
Thanks so much for this.
Will look into it right away.

**Comment by djb-rwth** [comment]
> There are examples of using this in Azure DevOps and GitHub actions on the main page of their CLI tool https://github.com/dotnet/sign

Hi @MikeWilliams-UK,
Unfortunately `.NET Foundations` is closed for new applications until next year.
Any further suggestions in this regard would be much appreciated.

**Comment by djb-rwth** [comment]
Hi all,
All `InChI` code will be digitally signed by [SignPath](https://about.signpath.io/) soon, and this issue will remain open until the mentioned digital signatures are applied.
Meanwhile, the code/binaries will be not be digitally signed in order to avoid [SmartScreen warning](https://github.com/IUPAC-InChI/InChI/issues/54#issuecomment-2470091557) reported by @MikeWilliams-UK.

**Comment by MikeWilliams-UK** [comment]
> > There are examples of using this in Azure DevOps and GitHub actions on the main page of their CLI tool https://github.com/dotnet/sign
> 
> Hi [@MikeWilliams-UK](https://github.com/MikeWilliams-UK), Unfortunately `.NET Foundations` is closed for new applications until next year. Any further suggestions in this regard would be much appreciated.

Just had latest newsletter from .NET Foundation and it says that "Membership Applications Are Open!"

https://dotnetfoundation.org/projects/submit-a-project

**Comment by fbaensch-beilstein** [comment]
@djb-rwth Can this be closed now with the integration of SignPath?

**Comment by MikeWilliams-UK** [comment]
@djb-rwth - Nope this can't be closed as I have just downloaded the latest (V1.07.4) and the binary is not signed.

<img width="692" height="520" alt="Image" src="https://github.com/user-attachments/assets/d0ca7592-06d2-40c7-b3c5-34bbfc89a3ea" />

**Comment by MikeWilliams-UK** [comment]
Nor is inchi-1.exe

<img width="681" height="514" alt="Image" src="https://github.com/user-attachments/assets/266e1090-1d00-4e40-bf09-aa2ff507fba3" />

**Comment by djb-rwth** [comment]
Hi @MikeWilliams-UK and @fbaensch-beilstein,
Unfortunately, _SignPath_ code signing has still not been applied due to a delayed ending of  #136 poll.
Required GitHub action was completed by @JanCBrammer though and just needs to be deployed.

More updates to follow.

**Comment by fbaensch-beilstein** [comment]
@djb-rwth Is this fixed with release v1.07.5?

**Comment by djb-rwth** [comment]
Hi @MikeWilliams-UK and @fbaensch-beilstein,
Executables are now signed with Microsoft code signing certificate.
